cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup.go

```diff
@@ -103,8 +112,10 @@ func (ng *nodegroup) DeleteNodes(nodes []*corev1.Node) error {
 	// and we fail fast.
 	replicas := ng.scalableResource.Replicas()
 
-	if replicas-int32(len(nodes)) <= 0 {
-		return fmt.Errorf("unable to delete %d machines in %q, machine replicas are <= 0 ", len(nodes), ng.Id())
+	if !ng.scalableResource.CanScaleFromZero() {
```

Comment:
`if ng.scalableResource.CanScaleFromZero()` is true we still  want to validate and error `if replicas-int32(len(nodes)) < 0 {`

Reply:
https://github.com/openshift/kubernetes-autoscaler/pull/137/files#r396385396

Reply:
seems like we need to check both conditions depending on if it can scale or not. eg:

```
if CanScaleFromZero and replicas-len(nodes) < 0
then
  error, deleting too many
else if !CanScaleFromZero and replicas-len(nodes) <= 0
then
  error, deleting too many
```